api/src/routes/auth.js

```diff
@@ -26,7 +28,12 @@ router.post('/login', (req, res) => {
   const { username, password } = req.body;
 
   User.findByUsername(username, (err, user) => {
-    if (err || !user || !bcrypt.compareSync(password, user.password)) {
+    if (err) {
+      console.error("Error in User.findByUsername:", err);
+      return res.status(500).send({ error: 'Internal Server Error' });
+    }
+    if (!user || !bcrypt.compareSync(password, user.password)) {
+      console.log("Invalid username or password.");
```

Comment:
Remove all console.log from file please

Reply:
I think we should keep it, we could create a ticket later for replacing the console log with actual an actual response that the api would give. The console log can stay as this `if` would be useless and achieve nothing when there is no content inside of here

Reply:
Yea, sending an error and a status code would be idea. I can change it but we should make tickets to see realistically what status codes we should be sending as well

Reply:
Idk maybe not a status code in some cases since some are just error checks. But definitely the error.

Reply:
yeyeye